src/Aspire.Hosting.Azure.CosmosDB/AzureCosmosDBResource.cs

```diff
@@ -14,15 +16,27 @@ public class AzureCosmosDBResource(string name, Action<AzureResourceInfrastructu
     IResourceWithConnectionString,
     IResourceWithEndpoints
 {
-    internal List<string> Databases { get; } = [];
+    internal List<CosmosDBDatabase> Databases { get; } = [];
 
     internal EndpointReference EmulatorEndpoint => new(this, "emulator");
 
     /// <summary>
     /// Gets the "connectionString" reference from the secret outputs of the Azure Cosmos DB resource.
+    ///
+    /// This is used when Entra ID authentication is used. The connection string is an output of the bicep template.
     /// </summary>
     public BicepSecretOutputReference ConnectionString => new("connectionString", this);
 
+    /// <summary>
+    /// Gets the "connectionString" secret output reference from the bicep template for the Azure Redis resource.
+    ///
+    /// This is set when access key authentication is used. The connection string is stored in a secret in the Azure Key Vault.
+    /// </summary>
+    internal BicepSecretOutputReference? ConnectionStringSecretOutput { get; set; }
+
+    [MemberNotNullWhen(true, nameof(ConnectionStringSecretOutput))]
+    internal bool UseAccessKeyAuthentication => ConnectionStringSecretOutput is not null;
```

Comment:
Maybe this should be public too.

Reply:
Checking the other resources with the same model (Redis, Postgres) they don't expose these publicly. Instead the `ConnectionStringExpression` is available and returns the one that corresponds to managed identity / access key.

I don't see how CosmosDB is different. (the emulator is equivalent to redis/postgres local containers, and the three have an Azure resource).